voter_info/frontend/public/index.html

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
```

Comment:
Honest question, not rhetorical: I'm curious what your thinking was on creating a separate `frontend` app, instead of putting the frontend files into `voter_info`. A `django` "app" is meant to hold models, views, routing, etc. for a single vertical bit of functionality. As the site we are building is (IIRC) a single-page app with one purpose (address -> exec branch officials), my first thought was a single django app.

i.e. `voter_info/static/...` for static html/css (https://docs.djangoproject.com/en/2.0/howto/static-files/)
`voter_info/views.py` for the one view that loads all the js/CSS
`voter_info/urls.py` to create the one route (which you did :-)


This is an open discussion I'd like to have -- based on all the handful of different libraries and frameworks you've been able to throw together so quickly you may have a better grasp of the current frontend coolness. My gut is that it's a little overcomplicated for a one page app, but there could very well be great reasons for all of them.

Reply:
I can get a completely interactive frontend up in react in basically no time. If we just wanted to write html, css, javascript we could but I like to keep away from reinventing the wheel for a lot of this stuff. We might as well use a framework that lets us move fast. Basically I was able to get a react project off the ground and running, returning a lat/long for an address in about 5 minutes. It's either do something like that and produce a mature looking front end or write all of the interactions and styles ourselves.

Reply:
I'm fine trashing react too, but this basically gives us something looking a little better than jquery manipulating an html page.

Reply:
I'm 100% for throwing in react and other frameworks for responsiveness. Only asking about _where_ it's located in the folder structure.

Reply:
Oh, the folder structure was kinda odd. 

I 'm totally ok with moving it, I actually think I had it that way originally. If we did that I would just change where the react build is output, we could have it write into django's static directory. I did it this way mainly because I added the react app's output directory to django so they had to be siblings to get that to work.

Reply:
Were you thinking it should be `$git_root/frontend`? And then the built js/html should output into django's static directory?

Reply:
I'm not wed to anything here, but something like:

* django's default `$git_root/$project/$app/static/....` for the static index html file that loads the webpack build. I.e. for us, `/voter_info/voter_info/static/...`. This folder contains all the static frontend, which for us is just the simple one-page HTML file that loads in our CSS and JS bundles.
* a django-project-level folder for the react JS app, e.g. `$git_root/voter_info/assets/js`: Contains the project-wide JS source e.g. react components, tests, utils.
* a django-project-level folder for the bundles that `webpack` creates, e.g. `$git_root/voter_info/assets/bundles`, this contains the compiled bundles from `$git_root/voter_info/assets/js`.

thoughts? concerns?

Reply:
I want to reiterate that I'm just asking questions -- you're closer to it.